src/components/CriteriaProvider.tsx

```diff
@@ -61,13 +70,15 @@ function elmLibraryToCriteria(elm: ElmLibrary, custom = false): Criteria[] {
   });
 }
 
-function jsonToCriteria(rawElm: string): Criteria[] | undefined {
-  const elm = JSON.parse(rawElm);
-  if (!elm.library?.identifier) {
+function jsonToCriteria(rawCriteria: string): Criteria[] | undefined {
+  const criteria = JSON.parse(rawCriteria);
+  if (criteria.type) {
+    return [builderModelToCriteria(criteria)];
```

Comment:
Does this branch ever get called? Seems like this function is only used when uploading a file, and I don't expect we're going to upload the builder model directly

Reply:
you're right, builderModelToCriteria wouldn't handle the raw string anyway

Reply:
FYI lint is throwing an error here now become `builderModelToCriteria` is imported but never used

Reply:
Have to remove the function definition since it's not being used anymore and causing a lint error.